ofproto/ofproto-dpif-xlate.c

```diff
@@ -3172,6 +3173,31 @@ get_fdb_data(struct xport *port, struct eth_addr mac_addr,
 
     return 0;
 }
+
+static int32_t
+update_ip_mac_map_info(struct flow *flow,
+             struct ip_mac_map_info *ip_mac_map_info)
+{
+    if (!flow) {
+        return -1;
+    }
+
+    memcpy(ip_mac_map_info->src_mac_addr, flow->dl_src.ea, sizeof(ip_mac_map_info->src_mac_addr));
+    memcpy(ip_mac_map_info->dst_mac_addr, flow->dl_dst.ea, sizeof(ip_mac_map_info->dst_mac_addr));
+
+    //Program the entry only for an ARP response where we have valid IP's and MAC for both src and dst
+    if(flow->nw_src && flow->nw_src != INADDR_ANY && flow->nw_src != INADDR_LOOPBACK
```

Comment:
This might be easier to read if you did a little refactoring.

Define a utility function:
```c
static inline int32_t
valid_ip_addr(ovs_be2 nw_addr) {
    return
        (nw_addr && nw_addr != INADDR_ANY &&
         nw_addr != INADDR_LOOPBACK && nw_addr != 0xffffffff);
}
```
And program the entry thus:
```c
if (valid_ip_addr(flow->nw_src) && !eth_addr_is_broadcast(flow->dl_src) &&
    valid_ip_addr(flow->nw_dst) && !eth_addr_is_broadcast(flow->dl_dst)) {
    ip_mac_map_info->src_ip_addr.family = AF_INET;
    ip_mac_map_info->src_ip_addr.ip.v4addr.s_addr = flow->nw_src;

    ip_mac_map_info->dst_ip_addr.family = AF_INET;
    ip_mac_map_info->dst_ip_addr.ip.v4addr.s_addr = flow->nw_dst;
}
```